code/android-lifecycle-library/src/main/java/com/adobe/marketing/mobile/lifecycle/LifecycleExtension.java

```diff
@@ -0,0 +1,394 @@
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * LifecycleExtension class
+ *
+ * The responsibility of LifecycleExtension is to handle the calculation and population of a base set of data within
+ * the SDK. This data will consist of information about the lifecycle of the app involving launches, installs
+ * and upgrades.
+ *
+ * This extension handles two main scenarios:
+ * <ul>
+ * 		<li> Computing standard lifecycle sessions, usually consumed by the Analytics extension</li>
+ *  	<li> Computing the application launch/close XDM metrics, usually consumed by the Edge Network and related extensions</li>
+ * </ul>
+ */
+public class LifecycleExtension extends Extension {
+
+	private static final String SELF_LOG_TAG                = "LifecycleExtension";
+	private final NamedCollection lifecycleDataStore;
+	private final DeviceInforming deviceInfoService;
+	private final LifecycleState lifecycleState;
+	private final LifecycleV2Extension lifecycleV2;
+
+	/**
+	 * Constructor for the LifecycleExtension, must be called by inheritors.
+	 * It is called by the Mobile SDK when registering the extension and it initializes the extension and registers event listeners.
+	 *
+	 * @param extensionApi {@code ExtensionApi} instance
+	 */
+	protected LifecycleExtension(final ExtensionApi extensionApi) {
+		super(extensionApi);
+		lifecycleDataStore = getDataStore();
+		deviceInfoService = getDeviceInfoService();
+		lifecycleState = new LifecycleState(lifecycleDataStore, deviceInfoService);
+		lifecycleV2 = new LifecycleV2Extension(lifecycleDataStore, deviceInfoService, getApi());
+	}
+
+	/**
+	 * This constructor is intended for testing purposes.
+	 *
+	 * @param extensionApi {@code ExtensionApi} instance
+	 * @param namedCollection {@code NamedCollection} instance
+	 * @param deviceInfoService {@code DeviceInforming} instance
+	 */
+	@VisibleForTesting
+	protected LifecycleExtension(final ExtensionApi extensionApi,
+								 final NamedCollection namedCollection,
+								 final DeviceInforming deviceInfoService) {
+		super(extensionApi);
+		lifecycleDataStore = namedCollection;
+		this.deviceInfoService = deviceInfoService;
+		lifecycleState = new LifecycleState(lifecycleDataStore, deviceInfoService);
+		lifecycleV2 = new LifecycleV2Extension(lifecycleDataStore, deviceInfoService, getApi());
+	}
+
+	/**
+	 * This constructor is intended for testing purposes.
+	 *
+	 * @param extensionApi {@code ExtensionApi} instance
+	 * @param namedCollection {@code NamedCollection} instance
+	 * @param deviceInfoService {@code DeviceInforming} instance
+	 * @param lifecycleState {@code LifecycleState} instance
+	 * @param lifecycleV2Extension {@code LifecycleV2Extension} instance
+	 */
+	@VisibleForTesting
+	protected LifecycleExtension(final ExtensionApi extensionApi,
+								 final NamedCollection namedCollection,
+								 final DeviceInforming deviceInfoService,
+								 final LifecycleState lifecycleState,
```

Comment:
Can you abstract all the regular lifecycle logic into LifecycleV1Extension class to have clearer seperation?

For ex) 
lifecycleStart() should delegate to 
lifecycleV1.start() similar to lifecycleV2.start()